include/ofi_mr.h

```diff
@@ -141,4 +146,51 @@ int ofi_mr_map_verify(struct ofi_mr_map *map, uintptr_t *io_addr,
 		      size_t len, uint64_t key, uint64_t access,
 		      void **context);
 
+
+/*
+ * Memory registration cache
+ */
+
+struct ofi_mr_entry {
+	struct iovec			iov;
+	uint64_t			access;	/* TODO */
```

Comment:
In the GNI provider, I found it was difficult to handle registrations of different types within the same cache. For the GNI provider, we split those registrations into two different caches -- one for read only regions, and one for read-write. For providers such as the verbs provider, this could get trickier in the event of read-only local, read-write, and read-remote registrations. Merging a read-only with a read-write could result in unexpected behavior from the fabric if an application tries to land data into a merged entry where it might have been read-only for a specific region of that registration prior to the merge.

Reply:
It would be nice to test the expectations of these functions with a set of tests before they get committed. 

I wouldn't expect a READ-only registration from 0xa000 to 0xb000 to be merged with a 0xa800 to 0xc000 READ-WRITE registration.

Reply:
I think separate caches will end up being preferable as well.   I'm inclined to remove the field until it is actually used.